lib/ansible/module_utils/common/validation.py

```diff
@@ -402,6 +402,8 @@ def check_type_list(value):
     """
     if isinstance(value, list):
         return value
+    elif not allow_conversion:
+        raise TypeError('%s cannot be converted to a list' % type(value))
```

Comment:
I'm not sure this is the best error message for this case; no attempt has been made to convert to a list, so this error is raised regardless of whether what was supplied *can* be converted to a list.